pkg/windows/windows.go

```diff
@@ -797,15 +799,19 @@ func (vm *windows) newFileInfo(path string) (*payload.FileInfo, error) {
 func (vm *windows) removeHNSNetworks() error {
 	vm.log.Info("removing HNS networks")
 	// VIP HNS endpoint created by the operator is also deleted when the HNS networks are deleted.
-	cmd := "\"$net0 = Get-HnsNetwork | where { $_.Name -eq '" + BaseOVNKubeOverlayNetwork + "'};" +
-		"$net1 = Get-HnsNetwork | where { $_.Name -eq '" + OVNKubeOverlayNetwork + "'};" +
-		"$net0 | Remove-HnsNetwork;" +
-		"$net1 | Remove-HnsNetwork;\""
-	_, err := vm.Run(cmd, true)
-	// PowerShell returns error waiting without exit status or signal error when the HNS networks are removed.
+	baseOvnCmd := "\"Get-HnsNetwork | where { $_.Name -eq '" + BaseOVNKubeOverlayNetwork + "'}| Remove-HnsNetwork;\""
+	_, err := vm.Run(baseOvnCmd, true)
+	if err != nil {
+		return errors.Wrapf(err, "error removing %s required HNS networks", BaseOVNKubeOverlayNetwork)
+	}
+	if err = vm.Reinitialize(); err != nil {
+		return errors.Wrapf(err, "error reinitializing VM after removing %s HNS network", BaseOVNKubeOverlayNetwork)
+	}
+	ovnCmd := "\"Get-HnsNetwork | where { $_.Name -eq '" + OVNKubeOverlayNetwork + "'}| Remove-HnsNetwork;\""
+	_, err = vm.Run(ovnCmd, true)
```

Comment:
Following this [comment](https://github.com/openshift/windows-machine-config-operator/pull/603#discussion_r691295828), the OVNKubeOverlayNetwork removal is the one  that drops the connectivity; Consider moving `vm.Reinitialize()` after, so no need to [reinitialize SSH client should if executing a step after removing networks](https://github.com/openshift/windows-machine-config-operator/pull/603/files#diff-4d8fe3bb66a872f0e4a277f820d60d2efa722f09aa89516be74b98c06b57f3bdR317).

Reply:
@jrvaldes re-initializing the ssh client in between is necessary to fix the flake. Removal of the BaseOvnKubeOverlayNetwork network causes reconfigurations and results into the flake of skipping the removal of OvnKubeOverlayNetwork on some instances.
Although I could add another vm.Reinitialize() at the end to remove the comment if that's the consensus. I did not add it because this is the last step in deconfiguration.

Reply:
> Removal of the BaseOvnKubeOverlayNetwork network causes reconfigurations

Reconfigurations by what entiity?

additionally:
could this be done in a loop?

something like:
```
for _, network := range []string{OVNKubeOverlayNetwork,BaseOVNKubeOverlayNetwork} {
  //removal logic
}
vm.reinitialize()
```

Reply:
can be done in a loop with the vm.Reinitialize call placed in the loop. I am not a 100% sure what re-configurations happen but I am here referring to HNSEndpoint and other networks that get deleted as part of the BaseOVNKubeOverlayNetwork deletion.